### ISSUE TITLE
[Bug]: PC0030 with RecordRef.GetBySystemId()

### ISSUE THREAD

**Issue by y0m0** [ALCops/Analyzers#360]
### Analyzer

PlatformCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

PC0030

### Description

As per [PC0030 documention](https://alcops.dev/docs/analyzers/platformcop/pc0030/), I would expect a RecordRef.GetBySystemId to not trigger this diagnostic.

<img width="1045" height="245" alt="Image" src="https://github.com/user-attachments/assets/f7f17cf2-72da-44a8-a31d-353122b869f1" />

### AL code to reproduce

```al
    procedure Test()
    var
        RecRef: RecordRef;
        Guid: Guid;
    begin
        RecRef.Open(Database::Customer);
        RecRef.GetBySystemId(Guid); // This lines triggers PC0030
    end;
```

**Comment by Arthurvdv** [comment]
Thanks for the report, @y0m0. I'd actually push back here — this is working as intended. Let me explain the reasoning.

PC0030's goal is to flag read operations where the runtime loads **all** normal fields (including every table-extension field, causing extra SQL joins) when you likely only need a few. That cost applies just as much to `RecordRef` as it does to a regular `Record`.

The key point: `SetLoadFields` is available on `RecordRef` too, not only on `Record`:
👉 https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/recordref/recordref-setloadfields-method

So your sample can be written as:

```al
procedure Test()
var
    RecRef: RecordRef;
    Guid: Guid;
begin
    RecRef.Open(Database::Customer);
    RecRef.SetLoadFields(...); // load only the fields you need
    RecRef.GetBySystemId(Guid);
end;
```

The snippet doesn't show what you do with the `RecRef` afterwards, but if you go on to read a value out of it, that's exactly the situation PC0030 is meant to help with — load only the fields you'll actually touch.

One honest caveat: if you're using the `RecordRef` for truly generic processing where you genuinely don't know upfront which fields you'll access, `SetLoadFields` may not be practical. In that case PC0030 is `Info` severity, so you can suppress it on that line.

I'll leave this open for now — does the above match your scenario, or do you have a concrete case where `SetLoadFields` can't reasonably be applied? Happy to dig in further.

**Comment by y0m0** [comment]
I agree with your take, and I can see that you have already updated the documentation and removed/reworder the line where it said that the diagnostic wouldn't trigger for RecRef.
I believe you can close this.
Thanks for your time and the amazing job you are doing with ALCops 🙏
